src/geom/hvabline.jl

```diff
@@ -138,17 +138,15 @@ function render(geom::ABLineGeometry, theme::Gadfly.Theme, aes::Gadfly.Aesthetic
     # the line extends to the edges of the graph.
 
     if typeof(aes.y) <: Array{Function}
-        low, high = aes.xmin[1], aes.xmax[1]
+        lowx, highx = aes.xmin[1], aes.xmax[1]
     else
-        xextrema = extrema(aes.x)
-        yextrema = extrema(aes.y)
-        low = min(xextrema[1], yextrema[1])
-        high = max(xextrema[2], yextrema[2])
+        lowx, highx = extrema(aes.x)
     end
 
-    range = high-low
-    x0 = low-range
-    x1 = high+range
+    # extending the line to width 3 times x-range of data should be enough
+    rangex = highx - lowx
+    x0 = lowx - rangex
+    x1 = highx + rangex
```

Comment:
what do these changes have to do with making Unitful work?  have you done regression testing to make sure nothing has changed by omitting the y-axis extremes?

Reply:
@bjarthur you can't compare y with x if y and x have different units. I regression tested, and the only differences I saw on my machine were between .js.svg graphs, which is to be expected as I understand it.

Reply:
@bjarthur you can't compare y with x if y and x have different units. I regression tested, and the only differences I saw on my machine were between .js.svg graphs, which is to be expected as I understand it.